src/nbody_simulation_result.jl

```diff
@@ -0,0 +1,172 @@
+# SimulationResult sould provide an interface for working with properties of a separate particle
+# and with physical properties of the whole system.
+struct SimulationResult{sType <: NBodySystem}
+    solution::AbstractTimeseriesSolution
+    simulation::NBodySimulation{sType}
+end
+
+function Base.show(stream::IO, sr::SimulationResult)
+    print(stream, "N: ") 
+    show(stream, length(sr.simulation.system.bodies))
+    println(stream)
+    show(stream, sr.simulation.system)
+    print(stream, "Time steps: ") 
+    show(stream, length(sr.solution.t))
+    println(stream)
+    print(stream, "t: ") 
+    show(stream, sr.solution.t)
+end
+
+(sr::SimulationResult)(t) = return sr.solution(t)
+
+# Instead of treating NBodySimulation as a DiffEq problem and passing it into a solve method
+# it is better to use a specific function for n-body simulations.
+function run_simulation(s::NBodySimulation, alg_type=Tsit5(), args...; kwargs...)
+    solution = solve(ODEProblem(s), alg_type, args...; kwargs...)
+    return SimulationResult(solution, s)
+end
+
+# this should be a method for integrators designed for the SecondOrderODEProblem (It is worth somehow to sort them from other algorithms)
+function run_simulation(s::NBodySimulation, alg_type::Union{VelocityVerlet,DPRKN6,Yoshida6}, args...; kwargs...)
+    solution = solve(SecondOrderODEProblem(s), alg_type, args...; kwargs...)
+    return SimulationResult(solution, s)
+end
+
+
+function get_velocity(sr::SimulationResult, time, i=0)
+    if typeof(sr.solution[1]) <: RecursiveArrayTools.ArrayPartition
+        velocities = sr(time).x[1]
+        n = size(velocities, 2)
+        if i <= 0
+            return velocities[:, 1:end]
+        else
+            return velocities[:, i]
+        end
+    else
+        velocities = sr(time)
+        n = div(size(velocities, 2), 2)
+        if i <= 0
+            return velocities[:, n + 1:end]
+        else
+            return velocities[:, n + i]
+        end
+    end
+end
+
+function get_position(sr::SimulationResult, time::Float64, i=0)
```

Comment:
This kind of stuff is a little too strictly typed. Why would `Float32` time fail? Just duck type it.

http://www.stochasticlifestyle.com/type-dispatch-design-post-object-oriented-programming-julia/

Reply:
Thanks for the link! I think time behaves like a Real number duck.